nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/CopyAzureBlobStorage_v12.java

```diff
@@ -0,0 +1,388 @@
+        return context.getProperty(SOURCE_STORAGE_CREDENTIALS_SERVICE).asControllerService(AzureStorageCredentialsService_v12.class);
+    }
+
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String sourceContainerName = context.getProperty(SOURCE_CONTAINER).evaluateAttributeExpressions(flowFile).getValue();
+        final String sourceBlobName = context.getProperty(SOURCE_BLOB_NAME).evaluateAttributeExpressions(flowFile).getValue();
+        final String destinationContainerName = Optional.ofNullable(
+                context.getProperty(DESTINATION_CONTAINER).evaluateAttributeExpressions(flowFile).getValue()
+        ).orElse(sourceContainerName);
+        final String destinationBlobName = Optional.ofNullable(
+                context.getProperty(DESTINATION_BLOB_NAME).evaluateAttributeExpressions(flowFile).getValue()
+        ).orElse(sourceBlobName);
+
+        final boolean createContainer = context.getProperty(AzureStorageUtils.CREATE_CONTAINER).asBoolean();
+        final AzureStorageConflictResolutionStrategy conflictResolution = AzureStorageConflictResolutionStrategy.valueOf(context.getProperty(AzureStorageUtils.CONFLICT_RESOLUTION).getValue());
+
+        long startNanos = System.nanoTime();
+        try {
+            BlobServiceClient storageClient = getStorageClient(context, DESTINATION_STORAGE_CREDENTIALS_SERVICE, flowFile);
+            BlobContainerClient containerClient = storageClient.getBlobContainerClient(destinationContainerName);
+            if (createContainer && !containerClient.exists()) {
+                containerClient.create();
+            }
+
+            BlobClient blobClient = containerClient.getBlobClient(destinationBlobName);
+            Map<String, String> attributes = new HashMap<>();
+            applyStandardBlobAttributes(attributes, blobClient);
+            final boolean ignore = conflictResolution == AzureStorageConflictResolutionStrategy.IGNORE_RESOLUTION;
+
+            final BlobRequestConditions destinationRequestConditions = new BlobRequestConditions();
+
+            try {
+                if (conflictResolution != AzureStorageConflictResolutionStrategy.REPLACE_RESOLUTION) {
+                    destinationRequestConditions.setIfNoneMatch("*");
+                }
+
+                final AzureStorageCredentialsService_v12 fromCredentialsService = getCopyFromCredentialsService(context);
+                BlobServiceClient fromStorageClient = getStorageClient(context, SOURCE_STORAGE_CREDENTIALS_SERVICE, flowFile);
+                BlobContainerClient fromContainerClient = fromStorageClient.getBlobContainerClient(sourceContainerName);
+                BlobClient fromBlobClient = fromContainerClient.getBlobClient(sourceBlobName);
```

Comment:
For naming consistency, recommend replacing `from` with `source` to align with the property names.
```suggestion
                BlobServiceClient sourceStorageClient = getStorageClient(context, SOURCE_STORAGE_CREDENTIALS_SERVICE, flowFile);
                BlobContainerClient sourceContainerClient = sourceStorageClient.getBlobContainerClient(sourceContainerName);
                BlobClient fromBlobClient = fromContainerClient.getBlobClient(sourceBlobName);
```

Reply:
I have replaced all occurances now in 6caaee05da2888ad5f2a36d4b8a157a88331e927.